staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -975,6 +975,12 @@ func (c *Cacher) GuaranteedUpdate(
 
 // Count implements storage.Interface.
 func (c *Cacher) Count(pathPrefix string) (int64, error) {
+	if c.ready.check() {
+		cnt := c.watchCache.Count()
+		if cnt != 0 {
```

Comment:
now that we're guarding on `c.ready.check()`, I'm not sure we need the zero-length check any more... we might just be able to return `c.watchCache.Count()`

Reply:
+1